File: Estudos/AlgortimosDeBusca/semana-1/quick-sort/quick_sort.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr  # Caso base: lista já ordenada ou vazia
    
    pivot = arr[-1]  # Escolhe o pivô (último elemento)
    left = []
    right = []

    for i in range(len(arr) - 1):
        if arr[i] < pivot:
            left.append(arr[i])
        else:
            right.append(arr[i])

    return quick_sort(left) + [pivot] + quick_sort(right)


# Teste
arr = [3, 6, 8, 10, 1, 2, 1]
print(quick_sort(arr))  # Saída: [1, 1, 2, 3, 6, 8, 10]

File: Estudos/AlgDeBusca/semana-1/quick-sort/quick_sort.js
function quickSort(arr) {
    if (arr.length <= 1) {
        return arr; // Caso base: lista já ordenada ou vazia
    }
    
    const pivot = arr[arr.length - 1]; // Escolhe o pivô (último elemento)
    let left = [];
    let right = [];
    
    for (let i = 0; i < arr.length - 1; i++) {
        if (arr[i] < pivot) {
            left.push(arr[i]);
        } else {
            right.push(arr[i]);
        }
    }

    return [...quickSort(left), pivot, ...quickSort(right)];
}

const arr = [3, 6, 8, 10, 1, 2, 1];
console.log(quickSort(arr)); // Saída: [1, 1, 2, 3, 6, 8, 10]

